From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Thu, 27 Apr 2023 22:20:08 +0200
Subject: [PATCH] HACK: drop shebang magic

We replace the shebang later anyways, so make sure it's always the same.

Signed-off-by: Michael Olbrich <m.olbrich@pengutronix.de>
---
 src/installer/scripts.py | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/src/installer/scripts.py b/src/installer/scripts.py
index 7e3c8fcd070a..ba6ed5aec3b6 100644
--- a/src/installer/scripts.py
+++ b/src/installer/scripts.py
@@ -59,20 +59,7 @@ def _build_shebang(executable: str, forlauncher: bool) -> bytes:
     https://bitbucket.org/pypa/distlib/src/58cd5c6/distlib/scripts.py#lines-124
     """
     executable_bytes = executable.encode("utf-8")
-    if forlauncher:  # The launcher can just use the command as-is.
-        return b"#!" + executable_bytes
-    if _is_executable_simple(executable_bytes):
-        return b"#!" + executable_bytes
-
-    # Shebang support for an executable with a space in it is under-specified
-    # and platform-dependent, so we use a clever hack to generate a script to
-    # run in ``/bin/sh`` that should work on all reasonably modern platforms.
-    # Read the following message to understand how the hack works:
-    # https://github.com/pradyunsg/installer/pull/4#issuecomment-623668717
-
-    quoted = shlex.quote(executable).encode("utf-8")
-    # I don't understand a lick what this is trying to do.
-    return b"#!/bin/sh\n'''exec' " + quoted + b' "$0" "$@"\n' + b"' '''"
+    return b"#!" + executable_bytes
 
 
 class InvalidScript(ValueError):
